### DISCUSSION TITLE
A new organisation for `pyani`

### DISCUSSION THREAD

**Discussion by widdowquinn** [widdowquinn/pyani#324] [Q&A]
As noted by Titus, it would be good to move `pyani`'s codebase to a standalone organisation, dissociating it from my personal set of repositories.

We could put the repo under `sipbs-compbiol`, but that still associates it with the university. I'm inclined to set up a completely separate organisation so that the whole codebase can be handed off, if necessary. For that we need a name that isn't taken.

The obvious choice for an organisation name would be `pyani` - but someone has already registered that as a username, so it can't be an organisation. We need a different name.

I'm currently thinking `pyani-org` or something similar, but it doesn't exactly sound inspired. Please add suggestions below.

**Comment by baileythegreen** [comment]
I actually don't think it makes sense to put `pyani` in the name of an organisation like this. I would use the organisation as a level of abstraction; use a name that is not tied to one tool in particular, because then you allow the possibility of housing other tools under its umbrella. Currently the only thing that might eventually also be there is `pdp`, I guess, but that can change.

I agree with not aligning it to the university; I will think about actual suggestions for the name.

**Comment by widdowquinn** [comment]
Thanks - I look forward to the suggestions.

I don't think I want to have an umbrella organisation with multiple tools, in this instance. I had imagined this organisation would cover only pyani, its documentation, and website (and whatever else comes up tied specifically to pyani).

The motivation is that the tool itself has become generally useful, and to have it tied to my own set of repositories is a hindrance to moving it on to the community (should that be a useful thing to do; could help avoid the bus problem to an extent, too). I don't know that the community would thank us for saddling them with a bunch of additional tools along with it ;)

**Reply by baileythegreen** [reply]
Some suggestions. I'm still thinking about this, so there may be more.

• **pANIply** — a play on 'panoply'[^1], as in, 'pyani offers a panoply of ways to run ANI'
• **orgANI** — pretty self-explanatory, I think[^2]
• **mANIc** — microbial ANI clique[^3]/crowd/covey/crew/club
• **mANIa** — microbial ANI association

It doesn't have to have ANI in the name, I know; but moving away from that I haven't yet found anything I thought worked well.

You'll notice none of my suggestions mention the language. I don't think the name of the organisation should; someone, at some point, might decide to port it to a different language, but that would be a bit weird if the organisation name mentioned Python.

[^1]: Yes, I was listening to The Decemberists when I came up with this; why do you ask? (Bonus points if you know the song in question.)
[^2]: See also variants on organism/organic/organise.
[^3]: Obviously the best choice in this list.

**Reply by widdowquinn** [reply]
ATM I seeing the purpose of the organisation being to house `pyani` and dissociate it from one individual (i.e. me) or organisation (i.e. my employer - despite the need for the copyright statement, etc.), so I'm inclined towards names that make this intuitively clear to the user. Prosaic as it is, something like `pyani-org` or `pyani-project` appeals to me more.

I don't know that porting to a different language is that likely to happen. One might argue that, by implementing the same analyses as `JSpecies`, `pyani` is a "port" of `JSpecies` to Python - but I'd disagree. I think it's a reimplementation of the independently-published methods, in Python. I'd see reimplementation of some of the measures in another language (with different interface) as being much more likely than porting.

**Reply by baileythegreen** [reply]
> I'm currently thinking `pyani-org` or something similar, but it doesn't exactly sound inspired. Please add suggestions below.

I was trying to give some suggestions that might count as 'inspired'. If you're still wanting to call it `pyani-org` or something similar, maybe just do it.

**Comment by baileythegreen** [comment]
I think moving `pyani` to a separate organisation will help with an issue I seem to be having. I can't view reports from CodeCov (not including what gets generated for a commit to a PR). I get a `403 Permission Error` and [this link](https://docs.codeclimate.com/docs/im-seeing-a-403-permission-error) seems to indicate it is because the 'organisation' (I guess, your account here) only has one owner (and it's not me).

**Reply by widdowquinn** [reply]
There is, at least, a workaround until we make the move - the coverage plugin should generate an equivalent report when pytest is run locally (as you probably already know).

**Reply by baileythegreen** [reply]
I don't, actually. Does this happen automatically? I was wanting to look at the coverage in individual files; I found that I can see the proportikn of each that is covered, but no more detail than that.

**Reply by widdowquinn** [reply]
Have a go with `pytest --cov-report=html --cov=pyani -v tests/` (there's a wee description in https://github.com/widdowquinn/pyani/blob/master/CONTRIBUTING.md#local-testing).
